crates/nix_rs/src/flake/outputs.rs

```diff
@@ -6,35 +6,69 @@ use std::{
     fmt::Display,
 };
 
-/// Absolute path to the `nix` binary compiled with flake schemas support
+/// Flake URL of the default flake schemas
 ///
 /// We expect this environment to be set in Nix build and shell.
-pub const NIX_FLAKE_SCHEMAS_BIN: &str = env!("NIX_FLAKE_SCHEMAS_BIN");
+pub const DEFAULT_FLAKE_SCHEMAS: &str = env!("DEFAULT_FLAKE_SCHEMAS");
 
-/// Flake URL of the default flake schemas
+/// Flake URL of the flake that defines functions for inspecting flake outputs
 ///
 /// We expect this environment to be set in Nix build and shell.
-pub const DEFAULT_FLAKE_SCHEMAS: &str = env!("DEFAULT_FLAKE_SCHEMAS");
+pub const INSPECT_FLAKE: &str = env!("INSPECT_FLAKE");
 
 /// Represents the "outputs" of a flake
 ///
-/// This structure is currently produced by `nix flake show`, thus to parse it we must toggle serde untagged.
+/// TODO: Rename this to `FlakeSchema` while generalizing the existing `schema.rs` module.
 #[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
-pub enum FlakeOutputs {
+pub struct FlakeOutputs {
+    pub inventory: BTreeMap<String, InventoryItem>,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum InventoryItem {
     Leaf(Leaf),
-    Attrset(BTreeMap<String, FlakeOutputs>),
+    Attrset(BTreeMap<String, InventoryItem>),
 }
 
 impl FlakeOutputs {
     /// Run `nix flake show` on the given flake url
     pub async fn from_nix(
         nix_cmd: &crate::command::NixCmd,
         flake_url: &super::url::FlakeUrl,
+        system: &super::System,
     ) -> Result<Self, crate::command::NixCmdError> {
-        let v = FlakeOutputsUntagged::from_nix(nix_cmd, flake_url).await?;
-        Ok(v.into_flake_outputs())
+        let v = nix_cmd
+            .run_with_args_expecting_json(&[
+                "eval",
+                "--json",
+                "--override-input",
+                "flake-schemas",
+                env!("DEFAULT_FLAKE_SCHEMAS"),
+                "--override-input",
+                "flake",
+                flake_url,
+                "--override-input",
+                "systems",
+                // TODO: Move `SystemsListFlakeRef` to `nix_rs` and use it here.
```

Comment:
Let me do this TODO once this PR is ready. I'll have to think about something here.